### PR TITLE
Ported `PythonGenerator.xtend` to Java and started refactoring of `GeneratorBase`

### PR DESCRIPTION
**Pull request by housengw**
See #838 for context.

Along with porting `PythonGenerator` to java, I also factored out part of the code into separate files (`PythonReactionGenerator`, `PythonParameterGenerator`, etc), which also led to factoring a small part of `CGenerator` into separate files (`CReactionGenerator`)

### CODE REVIEW

File: org.lflang/src/org/lflang/generator/python/PythonReactionGenerator.java [hunk 1]
```diff
@@ -99,13 +99,24 @@ private static String generateCPythonFunctionCaller(String reactorDeclName,
...
+     * 
+     * @param reaction The reaction to generate Python-specific initialization for.
+     * @param decl The reactor to which <code>reaction<code> belongs to.
+     * @param reactionIndex The index number of the reaction in decl.
+     * @param mainDef The main reactor.
+     * @param errorReporter An error reporter.
+     * @param types A helper class for type-related stuff.
+     * @param isFederatedAndDecentralized True if program is federated and coordination type is decentralized.
+     */
+    public static String generateCReaction(Reaction reaction,
```
Comment (Soroosh129) [inline org.lflang/src/org/lflang/generator/python/PythonReactionGenerator.java:113]: Would `generateCReactionBody` be better?

File: org.lflang/src/org/lflang/generator/python/PythonNetworkGenerator.java [hunk 2]
```diff
@@ -0,0 +1,122 @@
+package org.lflang.generator.python;
```
[the remark this answers is not in the document]
Comment (housengw) [reply org.lflang/src/org/lflang/generator/python/PythonNetworkGenerator.java:1]: The `generateNetwork*Body` functions is subject to refactor. My plan is to move all the federated network-related code to a file outside of `PythonGenerator.java`, be it `PythonGeneratorExtension.java` or `PythonNetworkGenerator.java`. I think the name `PythonNetworkGenerator.java` is more clear on the file's purpose

[the remark this answers is not in the document]
Comment (Soroosh129) [reply org.lflang/src/org/lflang/generator/python/PythonNetworkGenerator.java:1]: Sounds like a good plan. How about `PythonNetworkReactionGenerator.java` for the name?

[the remark this answers is not in the document]
Comment (housengw) [reply org.lflang/src/org/lflang/generator/python/PythonNetworkGenerator.java:1]: `PythonNetworkReactionGenerator.java` suggests that the code belongs in `PythonReactionGenerator.java`, I think `PythonNetworkGenerator.java` is more explicit about handling network-related stuff (like protobufs, serialization, etc)

**Review by lhstrh — APPROVED** [review]
This looks like a very sensible refactoring to me. Thank you for working on improving code quality so diligently!

**Pull request merged**